File: 백준/Gold/1300. K번째 수/K번째 수.py
import sys
input = sys.stdin.readline

size = int(input()) 
find_num = int(input())

start = 1
end = find_num
result = 0
while start <= end:
    mid = (start + end) // 2
    cnt = 0
    for i in range(1, size + 1):
        cnt += min(size, mid // i)
    if cnt < find_num:
        start = mid + 1
    else:
        result = mid
        end = mid - 1

print(result)

File: 백준/Gold/1300. K번째 수/K번째 수.cc
#include <iostream>
using namespace std;

int main() {
    int size, find_num;
    cin >> size >> find_num;

    int start = 1;
    int end = find_num;
    int result = 0;
    while(start <= end) {
        int mid = (start + end) / 2;
        int cnt = 0;
        for(int i = 1; i <= size; i++)
            cnt += min(size, mid / i);
        if (cnt < find_num)
            start = mid + 1;
        else {
            result = mid;
            end = mid - 1;
        }
    }
    
    cout << result << endl;

    return 0;
}